[vcpkg manifest] Add documentation! (#13488)
microsoft/vcpkg: docs/maintainers/maintainer-guide.md

Conflicting libs are typically by design and not considered a defect.  Because some build systems link against everything in the lib directory, these should be moved into a subdirectory named `manual-link`.

## Manifests and CONTROL files

When adding a new port or modifying an existing port, switch to the new manifest syntax for defining a port.

> You may do so easily by running the `vcpkg x-format-manifest` command, which will convert existing CONTROL

Reviewer: Again we should not be documenting `x-` commands.

Author: I think the right answer here is that `format-manifest` could be a non-`x-` command

Reviewer: Making it a non-X command sounds good to me!

Author: Did I not push..?

files into manifest files. Do not convert CONTROL files that have not been modified.

## Versioning

### Follow common conventions for the `"version"` field

See our [manifest files document](manifest-files.md#version-fields) for a full explanation of our conventions.

### Update the `"port-version"` field in the manifest file of any modified ports

Vcpkg uses this field to determine whether a given port is out-of-date and should be changed whenever the port's behavior changes.

> Our convention is to use the `"port-version"` field for changes to the port that don't change the upstream version, and to reset the `"port-version"` back to zero when an update to the upstream version is made.

Reviewer: ```suggestion
Change the `"port-version"` field for changes to the port that don't change the upstream version and require a port be reinstalled on upgrade, and to reset the `"port-version"` back to zero when an update to the upstream version is made.

```

It's not really a convention anymore now that it's a real field, and we want to discourage bumping that for something like adding a comment

Author: That's not a thing we've actually said though; at this point, the convention is to rev the port version for any changes (except on a case-by-case basis, like "reformatting"). If we want to change this convention, that's fine, but I don't think it should be decided in the comments here.
